lib/integrations/externalauditstorage/configurator.go

```diff
@@ -202,7 +200,10 @@ func newConfigurator(ctx context.Context, spec *externalauditstorage.ExternalAud
 			"ExternalAuditStorage: configured integration %q does not appear to be an AWS OIDC integration",
 			oidcIntegrationName)
 	}
-	awsRoleARN := awsOIDCSpec.RoleARN
+	awsRoleARN, err := arn.Parse(awsOIDCSpec.RoleARN)
```

Comment:
**Note to reviewers:** I'm not sure how this extra ARN validation will affect any configs in the wild.

Would the credential cache even work if this is either empty or an invalid ARN?

Reply:
An invalid RoleARN should not be accepted by AWS, so I expect this change to be ok.

Reply:
@nklaassen?